modules/kafka/src/main/java/com/devonfw/module/consumer/KafkaMessageReceiver.java

```diff
@@ -0,0 +1,115 @@
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Optional;
+import java.util.Properties;
+
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.springframework.util.CollectionUtils;
+import org.springframework.util.ObjectUtils;
+import org.springframework.util.StringUtils;
+
+import com.devonfw.module.common.config.KafkaConsumerProperties;
+
+/**
+ * @author Ravichandran
+ *
+ */
+public class KafkaMessageReceiver {
+
+  /**
+   * This method is used to receiver message from the kafka broker based on topic names and group names. If groupName is
+   * not provided all the messages belongs to the topic names will be returned
+   *
+   * @param topicNames
+   * @param kafkaConsumerProperties
+   * @return
+   */
+  public ConsumerRecords<Object, Object> receiveMessage(List<String> topicNames,
+      KafkaConsumerProperties kafkaConsumerProperties) {
+
+    KafkaConsumer<Object, Object> kafkaConsumer = null;
+
+    try {
+
+      kafkaConsumer = new KafkaConsumer<>(setConsumerProperties(kafkaConsumerProperties));
+
+      checkPreconditions(topicNames, kafkaConsumerProperties);
+
+      kafkaConsumer.subscribe(topicNames);
+
+      return kafkaConsumer.poll(Duration.ofMillis(1000));
+
+    } finally {
+      Optional.ofNullable(kafkaConsumer).ifPresent(consumer -> consumer.close());
+    }
+
+  }
+
+  /**
+   * This method is used to validate the inputs.
+   *
+   * @param topicNames
+   * @param kafkaConsumerProperties
+   */
+  private void checkPreconditions(List<String> topicNames, KafkaConsumerProperties kafkaConsumerProperties) {
+
+    if (CollectionUtils.isEmpty(topicNames)) {
+      throwIllegalArgumentException("topic names cannot be empty");
+    }
+
+    if (ObjectUtils.isEmpty(kafkaConsumerProperties)) {
+      throwIllegalArgumentException("kafka consumer peoperties cannot be null");
+    }
+
+    if (StringUtils.isEmpty(kafkaConsumerProperties.getBootstrapservers())) {
+      throwIllegalArgumentException("boot strap servrs cannot be null");
+    }
+
+    if (StringUtils.isEmpty(kafkaConsumerProperties.getKeyDeSerializerClassConfig())) {
+      throwIllegalArgumentException("key deserializer class config cannot be null");
+    }
+
+    if (StringUtils.isEmpty(kafkaConsumerProperties.getValueDeSerializerClassConfig())) {
+      throwIllegalArgumentException("value deserializer class config cannot be null");
+    }
+
+  }
+
+  /**
+   * This method is used to throw throwIllegalArgumentException when payloads are null.
+   *
+   */
+  private void throwIllegalArgumentException(String message) {
+
+    throw new IllegalArgumentException(message);
+  }
+
+  /**
+   * This method is used to set consumer properties.
+   *
+   * @param kafkaConsumerProperties
+   * @return
+   */
+  private Properties setConsumerProperties(KafkaConsumerProperties kafkaConsumerProperties) {
```

Comment:
so if this method converts `KafkaConsumerProperties` to regular `Properties`:
* why is the method called `setConsumerProperties`? It is not a setter!
* why not instead add an `asProperties()` or `toProperties()` method directly `KafkaConsumerProperties`?